### ISSUE TITLE
client message: Can't detect VirtualBox because this is a 32-bit version of BOINC; to fix, please install a 64-bit version

### ISSUE THREAD

**Issue by Ageless93** [BOINC/boinc#2508]
```
04-May-2018 13:07:16 [---] Can't detect VirtualBox because this is a 32-bit version of BOINC; to fix, please install a 64-bit version.
```
That's on a system that doesn't have VirtualBox installed.
It was even the maiden time I installed BOINC on it. For testing the client/OS start-up message, I installed the 32bit client on 64bit Windows 7 SP1. 
It gives it on the version WITHOUT VirtualBox. Not only as a message in the start-up messages, but also as a notice.

Referencing comment given by @TheAspens in https://github.com/BOINC/boinc/pull/2388#issuecomment-376938278

> I think I didn't state my opinion clearly enough above. I think that only one of the two following is the correct thing to do:
>
> *  No message should be displayed
> *  A simple generic message is displayed. I recommend the message
>```
> VirtualBox not detected. If you want BOINC to be able to use VirtualBox, then please see https://useful.wiki.page.on.boinc.site
> ```
> You cannot give users a very prominent message stating that something is wrong and that they must take action when in fact that may not be true.

**Comment by RichardHaselgrove** [comment]
That's fine: by all means improve the language and usability. But be explicit about what the real problem is:

```
08/05/2018 20:54:26 |  | Starting BOINC client version 7.10.2 for windows_intelx86
08/05/2018 20:54:28 |  | VirtualBox version: 5.2.8
```
BOINC can't cope with VirtualBox **IN MIXED MODE** - 32-bit BOINC on 64-bit OS. That's where we went wrong with the previous attempt to re-write that message, and why we reverted to the previous behaviour, but with the explicit identification that this is 32-bit BOINC, with VBox, on 32-bit Windows 7. It works.

**Comment by Ageless93** [comment]
> BOINC can't cope with VirtualBox IN MIXED MODE - 32-bit BOINC on 64-bit OS.

Not a problem in my case. I get the message with a 32bit BOINC on a 64bit Windows **without** VirtualBox.

So in that case, there should be extra detection for the 32bit Windows version of BOINC only:
- Is VirtualBox installed? 
-- If it isn't, all is OK, let's continue as is.
- If VirtualBox is installed, is it the 32bit version or 64bit version?
-- If it is the 64bit version, let's continue as is. 
- If it is the 32bit version, what bitness is the Windows we run on? 
-- If Windows is 32bit and VirtualBox is 32bit, let's continue.
- If Windows is 64bit and VirtualBox is 32bit, then chirp the message that this VirtualBox version cannot work on this 64bit Windows. 

Only in the last case do we need a message. Not in all cases where it's a 32bit BOINC with a 64bit Windows.

**Comment by RichardHaselgrove** [comment]
I'm pretty sure that VirtualBox can only be installed on the matching version of the OS, so your final bullet point (Windows is 64bit and VirtualBox is 32bit) can't exist.

Similarly, if this is a 32-bit version of windows, then VBox (if installed) will be 32-bit too - but only the 32-bit version of BOINC can tell if it's there or not.

The only possible case where we have a problem is 32-bit BOINC on 64-bit Windows - and in that case, we have no way of knowing whether VBox is installed or not.

**Comment by RichardHaselgrove** [comment]
You're not getting a message about needing to install VirtualBox. You're getting a message about needing to install BOINC - the 64-bit version - so that BOINC can assess the situation properly.

So, we have two problems.

1) It's a clumsy and badly-written message, and a notice as well - that might be overkill. But it comes from bc2ea6ede9a61e6d549d7dce36521cd8617b534f of Sep 23, 2014 - not from 7.8, but from [7.4.24](http://boinc.berkeley.edu/dev/forum_thread.php?id=8378&postid=57198)

2) We are likely to be advising more people to be running 32-bit BOINC on 64-bit Windows because of the as-yet unresolved #2470 - and that isn't going to be resolved until v7.12, as both @JuhaSointusalo and @TheAspens have said in 2470.

So, is a bad message which has survived through two previous versions a show-stopper this time? In my judgement. no, but I hope other people will express an opinion too.

**Comment by AenBleidd** [comment]
BOINC is released now as x64 only so this issue is not relevant anymore
